src/error.rs

```diff
@@ -181,12 +200,42 @@ impl Debug for ErrorImpl {
             }
             .finish()
         } else {
-            let () = match self {
+            let backtrace = match self {
                 #[cfg(feature = "dwarf")]
-                Self::Dwarf { error } => write!(f, "Error: {error}")?,
-                Self::Io { error } => write!(f, "Error: {error}")?,
-                Self::ContextOwned { context, .. } => write!(f, "Error: {context}")?,
-                Self::ContextStatic { context, .. } => write!(f, "Error: {context}")?,
+                Self::Dwarf {
+                    error,
+                    #[cfg(feature = "backtrace")]
+                    backtrace,
+                } => {
+                    let () = write!(f, "Error: {error}")?;
+                    #[cfg(feature = "backtrace")]
+                    {
+                        Some(backtrace)
+                    }
+                    #[cfg(not(feature = "backtrace"))]
+                    None
```

Comment:
Have you considered to have a separate method that would return just Option<Backtrace>? Then you can fetch that in this formatting method, but also it could be used in other contexts, if necessary. And formatting code will be less littered with `#cfg[]` alternatives?

Reply:
Yeah, but I couldn't make it work. The problem is that in order to work in both cases (both with and without backtraces) the method needs to have a backtrace argument as input. But we can't just generate such an object out of thin air if backtraces are disabled, right? Perhaps I am misunderstanding what you mean.

Reply:
So the issue is that we don't even have std::backtrace::Backtrace type defined, right? Kind of sucks...

How about this hack. Instead of using directly `std::backtrace::Backtrace`, you define your own newtype, and then depending on `#[cfg]` it will contain either `std::backtrace::Backtrace` or nothing/some dummy thing?

Well, anyway, if it's not going to spread beyond this code, it's probably fine.

Reply:
The type is present, it's more that we wouldn't have an object. So if we were to pass an `Option<Backtrace>` to the function, we are back at having conditional compilation just to conjure up this `Option` object. Will sleep over it and then decide what to do.

Reply:
Found a way now.